packages/engine/src/adapters/sql/index.ts

```diff
@@ -657,12 +725,26 @@ export class SQLAdapter implements GenericAdapter {
     return result as unknown as T
   }
 
-  async create<T>(tableName: string, data: Partial<T>): Promise<T> {
+  async create<T extends TableRecord<T>>(tableName: string, data: Partial<T>): Promise<T> {
+    const currentModel = this.databaseObjectionModel[tableName]
     adapterLogger.debug('create', tableName, data)
-    adapterLogger.debug('create', this.databaseObjectionModel[tableName])
-    adapterLogger.debug(this.databaseObjectionModel[tableName])
+    adapterLogger.debug('create', currentModel)
+    adapterLogger.debug(currentModel)
 
-    const query = this.databaseObjectionModel[tableName].query(this.database).insert(data)
+    /**
+     * Check if any data inserted is a geom one
+     */
+    const geomTables = currentModel.geomTables
+    Object.keys(data).forEach((dataKey) => {
+      if (geomTables.includes(dataKey)) {
+        // @ts-ignore
+        console.log('dataKey geom found ', dataKey, data?.[dataKey])
+        // @ts-ignore
+        data[dataKey] = raw(`ST_GeomFromText(?)::geometry`, [data[dataKey]])
```

Comment:
Is `::geometry` really necessary? The call to `ST_GeomFromText()` is not sufficient?

:warning: Some `console.log()` calls still remain in this function.

Reply:
You're right, I remove the `::geometry` cast.